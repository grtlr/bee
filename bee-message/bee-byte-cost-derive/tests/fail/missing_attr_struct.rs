// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#![allow(unused_imports, unreachable_patterns)]

use bee_byte_cost::ByteCost;

#[derive(Default, ByteCost)]
struct SingleField {
    a: u32,
}

fn main() {}
