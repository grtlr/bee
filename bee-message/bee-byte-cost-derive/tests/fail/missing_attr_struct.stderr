error[E0277]: the trait bound `u32: ByteCost` is not satisfied
  --> tests/fail/missing_attr_struct.rs:10:5
   |
10 |     a: u32,
   |     ^ the trait `ByteCost` is not implemented for `u32`
   |
note: required by `weighted_bytes`
  --> $WORKSPACE/bee-message/bee-byte-cost/src/lib.rs
   |
   |     fn weighted_bytes(&self, config: &ByteCostConfig) -> u64;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
